### DISCUSSION TITLE
Error without GPU

### DISCUSSION THREAD

**Discussion by sakerIN** [lllyasviel/ControlNet#340] [Q&A]
Does this model work if we only have cpu? We've tried, but it showed an error as picture below. Any suggenstions to solve this problem?

![image](https://user-images.githubusercontent.com/106565411/231641035-7d208c5c-c1d4-48db-a2cc-f64ff128523b.png)

**Comment by My12123** [comment]
This is because you have a [PyTorch](https://pytorch.org/get-started/previous-versions) version for CUDA. You need a CPU version.
![image](https://user-images.githubusercontent.com/49092250/231701580-dfdc5b57-57aa-420a-8944-963c1f4458a1.png)
First remove the version for CUDA and then install the CPU version with the commands `pip uninstall torch` 
for Windows `pip3 install torch torchvision torchaudio` for Linux `pip3 install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cpu`

**Reply by sakerIN** [reply]
This I have considered, I install is the cpu version

![image](https://user-images.githubusercontent.com/106565411/231712560-7b508b83-7aa0-4058-959f-33f896f36c33.png)

**Reply by sakerIN** [reply]
I understand that this is a problem in the code. I have changed part of cuda suffix to cpu, but this error is still reported

![image](https://user-images.githubusercontent.com/106565411/231713186-faa3d63e-174c-4530-92d8-4673b7e2e71b.png)
